doc/source/whatsnew/v2.1.0.rst

```diff
@@ -304,6 +304,7 @@ Deprecations
 - Deprecated literal string/bytes input to :func:`read_html`. Wrap literal string/bytes input in ``io.StringIO`` / ``io.BytesIO`` instead. (:issue:`53767`)
 - Deprecated option "mode.use_inf_as_na", convert inf entries to ``NaN`` before instead (:issue:`51684`)
 - Deprecated parameter ``obj`` in :meth:`GroupBy.get_group` (:issue:`53545`)
+- Deprecated parsing datetimes with mixed time zones unless user passes ``utc=True`` to :func:`to_datetime` (:issue:`50887`)
```

Comment:
Could you describe how a user could get the old behavior (here and in the warning message)?